src/components/dashboard/main-content/course-enrollments/CourseEnrollments.jsx

```diff
@@ -110,11 +116,16 @@ const CourseEnrollments = ({ children }) => {
     );
   }
 
+  const handleOnClose = () => {
+    setShowCancelledAssignmentsAlert(false);
+    global.localStorage.setItem(LEARNER_ACKNOWLEDGED_CANCELLATION_ALERT, true);
```

Comment:
I'm wondering if we need to consider a case where the user has dismissed the cancellation alert but gets new cancelled assignments. In that scenario, I think we also need to keep track of the cancelled assignment uuid's in local storage and reconcile against it.

Reply:
Yeah, great callout; I agree just storing the boolean in localStorage would prevent it from showing for future cancelations after the initial "Dismiss".

Keeping track of the dismissed assignment uuids in localStorage could work. One alternative (related to / adapted from one of Alex D.'s earlier ideas): would it make sense to store the timestamp of the "Dismiss" click in localStorage and then only show cancelations that are more recent / beyond that stored timestamp as the "new" canceled assignments to display? The next "Dismiss" click would overwrite the saved timestamp in localStorage as the next reference timestamp.

Either approach could likely work, though the date filtering would likely be a bit more performant (i.e., less array iteration needed to filter/exclude the already-dismissed canceled assignments).

Reply:
I went with the approach to add a timestamp in local storage!